eZ/Bundle/EzPublishIOBundle/Command/MigrateFilesCommand.php

```diff
@@ -0,0 +1,276 @@
+        }
+
+        $bulkCount = $input->getOption('bulk-count');
+        $dryRun = $input->getOption('dry-run');
+
+        $this->migrateFiles(
+            $totalCount,
+            $bulkCount,
+            $dryRun,
+            $output
+        );
+    }
+
+    /**
+     * Output the configured meta/binary data handlers.
+     *
+     * @param OutputInterface $output
+     */
+    protected function outputConfiguredHandlers(OutputInterface $output)
+    {
+        $output->writeln(
+            'Configured meta data handlers: ' . implode(', ', array_keys($this->configuredMetadataHandlers))
+        );
+        $output->writeln(
+            'Configured binary data handlers: ' . implode(', ', array_keys($this->configuredBinarydataHandlers))
+        );
+    }
+
+    /**
+     * Verify that the handler options have been set to meaningful values.
+     *
+     * @param mixed $fromHandlers
+     * @param mixed $toHandlers
+     * @param OutputInterface $output
+     * @return bool
+     */
+    protected function areHandlerOptionsValid(
+        $fromHandlers,
+        $toHandlers,
+        OutputInterface $output
+    ) {
+        if ($fromHandlers) {
+            if (count($fromHandlers) !== 2) {
+                $output->writeln('Enter two comma separated values for the --from option: <from_metadata_handler>,<from_binarydata_handler>');
+
+                return false;
+            }
+
+            if (!in_array($fromHandlers[0], array_keys($this->configuredMetadataHandlers))) {
+                $output->writeln("From meta data handler '$fromHandlers[0]' is not configured.");
+                $this->outputConfiguredHandlers($output);
+
+                return false;
+            }
+
+            if (!in_array($fromHandlers[1], array_keys($this->configuredBinarydataHandlers))) {
+                $output->writeln("From binary data handler '$fromHandlers[1]' is not configured.");
+                $this->outputConfiguredHandlers($output);
+
+                return false;
+            }
+        }
+
+        if ($toHandlers) {
+            if (count($toHandlers) !== 2) {
+                $output->writeln('Enter two comma separated values for the --to option: <to_metadata_handler>,<to_binarydata_handler>');
+
+                return false;
+            }
+
+            if (!in_array($toHandlers[0], array_keys($this->configuredMetadataHandlers))) {
+                $output->writeln("To meta data handler '$toHandlers[0]' is not configured.");
+                $this->outputConfiguredHandlers($output);
+
+                return false;
+            }
+
+            if (!in_array($toHandlers[1], array_keys($this->configuredBinarydataHandlers))) {
+                $output->writeln("To binary data handler '$toHandlers[1]' is not configured.");
+                $this->outputConfiguredHandlers($output);
+
+                return false;
+            }
+        }
+
+        if ($fromHandlers && $toHandlers && $fromHandlers === $toHandlers) {
```

Comment:
No, because I support @bdunogier's proposal that if you don't specify from/to they will be set to default handler and first configured handler, respectively. However if I do validation after that point, I could reduce the if like this indeed. And that's probably a good idea, because you could potentially configure your site so that the default from/to would end up being the same.